packages/ui/src/components/va-button/VaButton.vue

```diff
@@ -1,241 +1,211 @@
 
-      if (focusState.value) {
-        const alpha = props.outline ? -0.8 : -0.7
-        const lightness = 10
-        const color = isTransparentBackground.value ? shiftHSLAColor(colorComputed.value, { a: alpha }) : shiftHSLAColor(colorComputed.value, { l: lightness })
+    // styles
+    const isLowContrastBg = computed(() => props.plain || isLightBackground(colorComputed.value, props.backgroundOpacity))
+    const { textColorComputed } = useTextColor(colorComputed, isLowContrastBg.value)
 
-        background = props.gradient ? getGradientBackground(color) : color
-      }
+    const backgroundComputed = useButtonBackground(colorComputed, isPressed, isHovered)
+    const contentColorComputed = useButtonTextColor(textColorComputed, colorComputed, isPressed, isHovered)
 
-      return {
-        color: textColorComputed.value,
-        borderColor,
-        background,
-      }
-    })
+    const computedStyle = computed(() => ({
+      borderColor: props.borderColor ? getColor(props.borderColor) : 'transparent',
+      ...backgroundComputed.value,
+      ...contentColorComputed.value,
+    }))
 
-    const focus = () => button.value?.focus()
-    const blur = () => button.value?.blur()
+    const publicMethods = { focus, blur }
 
     return {
       button,
       tagComputed,
-      hrefComputed,
       computedClass,
       computedStyle,
-      computedType,
       textColorComputed,
-      loaderSize,
-      focusState,
-      hoverState,
-      focus,
-      blur,
+      loaderSizeComputed,
+      attributesComputed,
+      wrapperClassComputed,
+      iconAttributesComputed,
+
+      ...publicMethods,
     }
   },
 })
 </script>
 
 <style lang='scss'>
-@import '../../styles/resources';
 @import 'variables';
+@import '../../styles/resources';
 
 .va-button {
+  position: relative;
+  margin: var(--va-button-margin);
+  padding: var(--va-button-padding);
   display: var(--va-button-display);
-  align-items: var(--va-button-align-items);
   justify-content: var(--va-button-justify-content);
+  align-items: var(--va-button-align-items);
+  border: var(--va-button-border);
   background-image: var(--va-button-background-image);
-  box-shadow: var(--va-button-box-shadow, var(--va-control-box-shadow));
-  outline: var(--va-button-outline);
-  border: var(--va-button-border, var(--va-control-border));
+  box-shadow: var(--va-button-box-shadow);
   font-family: var(--va-font-family);
-  text-decoration: none !important;
+  font-weight: var(--va-button-font-weight);
+  text-decoration: none;
   text-transform: initial;
-  cursor: pointer;
-  transition: var(--va-button-transition, var(--va-swing-transition));
-  background-color: var(--va-button-background-color, var(--va-white));
-  vertical-align: middle;
+  transition: var(--va-button-transition);
   box-sizing: border-box;
-  font-weight: var(--va-button-font-weight);
-  margin: var(--va-button-margin);
-  padding: var(--va-button-padding);
-  position: relative;
+  cursor: pointer;
```

Comment:
We probably can move cursor to variable as well. This is a valid point for customization.